File: chapi/solutions/two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        new_nums = [[nums[i], i] for i in range(len(nums))]
        new_nums.sort()
        last = len(nums) - 1
        first = 0
        while last > first:
            s = new_nums[first][0] + new_nums[last][0]
            if s == target:
                return [new_nums[first][1], new_nums[last][1]]
            elif s > target:
                last -= 1
            else:
                first += 1
            
            

File: chapi/solutions/two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<pair<int, int> > vi;
        int c = 0;
        for(auto i: nums){
            vi.push_back({i, c++});
        }
        sort(vi.begin(), vi.end());
        int i = 0, j = nums.size() - 1;
        while( i < j){
            int res = vi[i].first + vi[j].first;
            if (res == target){
                i = vi[i].second, j = vi[j].second;
                break;
            } else if(res < target){
                i++;
            } else {
                j--;
            }
        }
        return {i,j};
    }
};